File: app/src/main/java/com/egoriku/ladyhappy/presentation/ui/base/BaseActivity.java
package com.egoriku.ladyhappy.presentation.ui.base;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.StringRes;

import com.egoriku.corelib_kt.arch.BaseContract;

public abstract class BaseActivity<V extends BaseContract.View, P extends BaseContract.Presenter<V>> extends com.egoriku.corelib_kt.arch.BaseActivity {

    public abstract void getExtras(Intent _intent);

    public abstract void setUpToolbar(@StringRes int title);

    public Context getContext() {
        return this;
    }
}

File: app/src/main/java/com/egoriku/ladyhappy/presentation/ui/base/BaseActivity.kt
package com.egoriku.ladyhappy.presentation.ui.base

import android.content.Context
import android.content.Intent
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity


abstract class BaseActivity : AppCompatActivity() {

    abstract fun getExtras(_intent: Intent)
    abstract fun setUpToolbar(@StringRes title:Int)
    abstract fun getAttachedFragment(id: Int): Fragment
    abstract fun getAttachedFragment(tag: String): Fragment

    fun getContext(): Context {
        return this
    }
}